app/models/katello/generic_content_unit.rb

```diff
@@ -0,0 +1,68 @@
+module Katello
+  class GenericContentUnit < Katello::Model
+    self.table_name = 'katello_generic_content_units'
+    include Concerns::PulpDatabaseUnit
+
+    CONTENT_TYPE = 'generic_content_unit'.freeze
+
+    def self.default_sort
+      order(:name)
+    end
+
+    def self.total_for_repositories(repos)
+      self.in_repositories(repos).count
+    end
+
+    def self.import_all(pulp_ids = nil, repository = nil)
+      ids_to_associate = []
+      service_class = SmartProxy.pulp_primary!.content_service(content_type)
+      service_class.pulp_units_batch_all(pulp_ids).each do |units|
+        units.each do |unit|
+          unit = unit.with_indifferent_access
+          if content_type == 'rpm' && repository
+            rpms_to_disassociate = ::Katello::Rpm.where(name: unit[:name], version: unit[:version], release: unit[:release],
+                                                        epoch: unit[:epoch], arch: unit[:arch]).select(:id)
+            if rpms_to_disassociate.any?
+              ::Katello::RepositoryRpm.where(rpm_id: rpms_to_disassociate, repository_id: repository.id).destroy_all
+            end
+          end
+          model = Katello::Util::Support.active_record_retry do
+            self.where(:pulp_id => unit[service_class.unit_identifier]).first_or_create
+          end
+          service = service_class.new(model.pulp_id)
+          service.backend_data = unit
+          service.update_model(model, repository.repository_type)
+          ids_to_associate << model.pulp_id
+        end
+      end
+      sync_repository_associations(repository, :pulp_ids => ids_to_associate, :additive => true) if self.many_repository_associations && repository && ids_to_associate.present?
+    end
+
+    def self.import_for_repository(repository)
+      pulp_id_href_map = {}
+      service_class = SmartProxy.pulp_primary!.content_service(content_type)
+      fetch_only_ids = !repository.content_view.default? &&
+        !repository.repository_type.unique_content_per_repo &&
+        service_class.supports_id_fetch?
+
+      service_class.pulp_units_batch_for_repo(repository, fetch_identifiers: fetch_only_ids).each do |units|
+        units.each do |unit|
+          unit = unit.with_indifferent_access
+          pulp_id = unit[service_class.unit_identifier]
+          backend_identifier = unit.dig(service_class.backend_unit_identifier)
+          unless fetch_only_ids
+            model = Katello::Util::Support.active_record_retry do
+              self.where(:pulp_id => pulp_id).first_or_create
+            end
+            service = service_class.new(model.pulp_id)
+            service.backend_data = unit
+            model.repository_id = repository.id unless many_repository_associations
+            service.update_model(model, repository.repository_type)
```

Comment:
We don't want to redefine the entirety of `import_for_repository` and `import_all` here because it makes the code hard to maintain.  Instead, I think you should use the methods that are included with `include Concerns::PulpDatabaseUnit`.

To make this work, I think you'll just need to add a conditional here to check if the backend service is generic and then send in the proper args.